### PR TITLE
feat: Tune strips on landing

### LINKED ISSUE #8830: Tune strips on landing
Let's tune it for better trading, especially first-timers

Generative strip
-
- expand it to at least 18 items from AHP

Latest Sales
-
- three item less from rmrk 
- three less from rmrk2 
show extra 
- three item from AHP
- three item from AHK

Newest list
-
- same as latest sales

image

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #8830

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI;

### CODE REVIEW

File: components/carousel/utils/useCarouselEvents.ts [hunk 1]
```diff
@@ -165,7 +173,8 @@ export const useCarouselNftEvents = ({ type }: Types) => {
   return computed(() => items.value.nfts)
 }
 
-const generativeLimit = 10
+const AHK_GENERATIVE_LIMIT = 10
+const AHP_GENERATIVE_LIMIT = 12
```
Comment (yangwao) [inline components/carousel/utils/useCarouselEvents.ts:177]:
```suggestion
const AHP_GENERATIVE_LIMIT = 18
```

File: components/carousel/utils/useCarouselEvents.ts [hunk 2]
```diff
@@ -165,7 +173,8 @@ export const useCarouselNftEvents = ({ type }: Types) => {
   return computed(() => items.value.nfts)
 }
 
-const generativeLimit = 10
+const AHK_GENERATIVE_LIMIT = 10
```
Comment (yangwao) [inline components/carousel/utils/useCarouselEvents.ts:176]:
```suggestion
const AHK_GENERATIVE_LIMIT = 6
```

File: components/carousel/CarouselTypeNewestList.vue [hunk 3]
```diff
@@ -9,8 +9,15 @@
...
+import { limit, useCarouselNftEvents } from './utils/useCarouselEvents'
 
 const { urlPrefix } = usePrefix()
-const nfts = useCarouselNftEvents({ type: 'newestList' })
+const nfts = useCarouselNftEvents({
+  type: 'newestList',
+  limit: {
+    ahp: limit + 1,
+    rmrk: limit - 1,
+    ksm: limit - 1,
```
Comment (yangwao) [inline components/carousel/CarouselTypeNewestList.vue:20]:
```suggestion
    ksm: limit - 3,
```

File: components/carousel/CarouselTypeLatestSales.vue [hunk 4]
```diff
@@ -8,8 +8,16 @@
...
 
 const { urlPrefix } = usePrefix()
-const nfts = useCarouselNftEvents({ type: 'latestSales' })
+
+const nfts = useCarouselNftEvents({
+  type: 'latestSales',
+  limit: {
+    ahp: limit + 1,
+    rmrk: limit - 1,
+    ksm: limit - 1,
```
Comment (yangwao) [inline components/carousel/CarouselTypeLatestSales.vue:20]:
```suggestion
    ksm: limit - 3,
```

File: components/carousel/CarouselTypeLatestSales.vue [hunk 5]
```diff
@@ -8,8 +8,16 @@
...
-import { useCarouselNftEvents } from './utils/useCarouselEvents'
+import { limit, useCarouselNftEvents } from './utils/useCarouselEvents'
 
 const { urlPrefix } = usePrefix()
-const nfts = useCarouselNftEvents({ type: 'latestSales' })
+
+const nfts = useCarouselNftEvents({
+  type: 'latestSales',
+  limit: {
+    ahp: limit + 1,
```
Comment (yangwao) [inline components/carousel/CarouselTypeLatestSales.vue:18]:
```suggestion
    ahp: limit + 3,
```

File: components/carousel/CarouselTypeNewestList.vue [hunk 6]
```diff
@@ -9,8 +9,15 @@
...
 <script lang="ts" setup>
-import { useCarouselNftEvents } from './utils/useCarouselEvents'
+import { limit, useCarouselNftEvents } from './utils/useCarouselEvents'
 
 const { urlPrefix } = usePrefix()
-const nfts = useCarouselNftEvents({ type: 'newestList' })
+const nfts = useCarouselNftEvents({
+  type: 'newestList',
+  limit: {
+    ahp: limit + 1,
```
Comment (yangwao) [inline components/carousel/CarouselTypeNewestList.vue:18]:
```suggestion
    ahp: limit + 3,
```

**Comment by prury** [comment]
Comparison: 
| Strip  | Canary | Deployed | 
| ------------- | ------------- | ------------- |
| Generative  | 16 ahp + 9 ahk | 16 ahp + 9 ahk | 
| Latest Sales  | 7 rmrk2 + 7 rmrk + 8 ahp | 7 rmrk2 + 6 rmrk + 7 ahp | 
| Newest List | 7 ahp + 7 rmrk2 + 8 rmrk | 7 ahp + 7 rmrk2 + 8 rmrk |

**Comment by yangwao** [comment]
> Comparison:

yeah by quick skimming noticed same, did some changes, wait for deploy

**Comment by prury** [comment]
> > Comparison:
> 
> yeah by quick skimming noticed same, did some changes, wait for deploy

canary already had more than 12 ahp on generative tho

**Comment by yangwao** [comment]
> canary already had more than 12 ahp on generative tho

yup goal would be at least three rolls best

cc @hassnian I've adjusted numbers bit, let's see, I think that limit doesn't work as we think

> three item less from rmrk and three less from rmrk2 and show three extra item from AHP, same for sales

**Comment by yangwao** [comment]
generative strip still 16 items hmm

**Comment by prury** [comment]
let's resolve merge conflicts because i still see some bsx items, i would like to check deployed without them

**Review by preschian — COMMENTED** [review]
I still can see items from rmrk/rmrk2 are more than > 3

I just noticed some part of the code missing on `useChainEvents`. Maybe missing out during migration nuxt3. checking

**Comment by hassnian** [comment]
> I still can see items from rmrk/rmrk2 are more than > 3
> 
> I just noticed some part of the code missing on `useChainEvents`. Maybe missing out during migration nuxt3. checking

mb, bad name, renamed it to limitDifference

ksm will have 3 items less and rmrk 1 less from the total limit per chain which is 8

**Comment by preschian** [comment]
Hi, sorry, I need to jump into this issue. I need to adjust it to something like this: https://github.com/kodadot/nft-gallery/pull/8922.

Some missing parts after migration. Old code here for comparison: https://github.com/kodadot/nft-gallery/blob/ca68c0c24b2358917bec4d141161bcf8888d7cab/components/carousel/utils/useCarouselEvents.ts#L94-L98

*ignore the builds in that PR. Seems like GitHub actions were down

**Comment by hassnian** [comment]
> Hi, sorry, I need to jump into this issue. I need to adjust it to something like this: #8922.

np 👍 


closing in favour of #8922

**Pull request closed without merging**